src/main/java/thaumicenergistics/common/inventory/HandlerKnowledgeCore.java

```diff
@@ -188,6 +188,8 @@ private void removePatternSameInputs(final ArcaneCraftingPattern pattern) {
 				continue;
 			boolean match = true;
 			for (int i = 0; i < newInputs.length; ++i) {
+		                if (null == oldInputs[i] || null == newInputs[i])
```

Comment:
shouldn't it change 'match' variable to false? I don't know the mechanic, asking just in case.
Also, could you please swap `null == oldInputs[i] || null == newInputs[i]` to `oldInputs[i] == null || newInputs[i] == null`

Reply:
Make new commit with changed behavior.